### PR TITLE
changed sds storage pool test and rfcache test to get list of parameters

### PR DESCRIPTION
**Pull request by tikitavi**

### CODE REVIEW

File: juju/fuel/functions [hunk 1]
```diff
@@ -213,54 +213,61 @@ function check_storage_pool {
...
   login_cmd="scli --login --username $username --password $password --approve_certificate"
-  query_cmd="scli --query_storage_pool --protection_domain_name $pd --storage_pool_name $sp"
-  if output=`juju ssh $mch "$login_cmd && $query_cmd" 2>/dev/null | grep "$key_to_check" | grep -q "$expected_value"` ; then
-    echo "INFO: Success. Parameter $key_to_check is $expected_value."
-  else
-    echo "ERROR: Parameter $key_to_check is in wrong state."
-    echo "$output"
-    return 1
-  fi
+  for sp in "${sp_list[@]}" ; do
```
Comment (Andrey-mp) [inline juju/fuel/functions]: for sp in `echo "$storage_pools" | sed 's/,/ /g'` ; do

and below

File: juju/fuel/deploy-fuel.sh [hunk 2]
```diff
@@ -28,27 +27,29 @@ check_storage_pool 1 'Flash Read Cache' "Doesn't use"
 
 remove_node_service 1 2 3
 set_fuel_options protection-domain='pd'
-set_fuel_options storage-pools='sp'
+set_fuel_options storage-pools='sp1,sp2'
+set_fuel_options device-paths='/dev/xvdf,/dev/xvdg'
```
Comment (Andrey-mp) [inline juju/fuel/deploy-fuel.sh]: extract disks definition into variable

File: juju/fuel/deploy-fuel.sh [hunk 3]
```diff
@@ -28,27 +27,29 @@ check_storage_pool 1 'Flash Read Cache' "Doesn't use"
...
-set_fuel_options rfcache-devices=$rfcache_path
+set_fuel_options cached-storage-pools='sp2'
+set_fuel_options rfcache-devices=$rfcache_paths
 configure_cluster mode 1 primary-controller 1 compute 2,3
 
+rfcache=`echo "$rfcache_paths" | sed 's/,/ /'`
 check_password 1 'Other_password'
 check_protection_domain 1 'pd'
-check_sp_name 1 'sp'
+check_sds_storage_pool 1 'sp1 sp2' '/dev/xvdf /dev/xvdg'
```
Comment (Andrey-mp) [inline juju/fuel/deploy-fuel.sh]: change call - pass parameters with ',' and do all with them inside function

File: juju/fuel/deploy-fuel.sh [hunk 4]
```diff
@@ -28,27 +27,29 @@ check_storage_pool 1 'Flash Read Cache' "Doesn't use"
 
 remove_node_service 1 2 3
 set_fuel_options protection-domain='pd'
-set_fuel_options storage-pools='sp'
+set_fuel_options storage-pools='sp1,sp2'
```
Comment (Andrey-mp) [inline juju/fuel/deploy-fuel.sh]: extract pools list into variable and use it here and below

File: juju/fuel/functions [hunk 5]
```diff
@@ -213,54 +213,61 @@ function check_storage_pool {
...
+  done
 }
 
-function check_path {
+function check_rfcache {
   mch=${machines[$1]}
-  path_type=$2
-  local device_path=$3
+  rfcache_list=($2)
+  rfcache_number=${#rfcache_list[@]}
```
Comment (Andrey-mp) [inline juju/fuel/functions]: number -> count

File: juju/fuel/functions [hunk 6]
```diff
@@ -332,12 +343,26 @@ function check_sp_name {
...
-    else
-      echo "INFO: Success. Storage pool name on $sds_name is $sp_name."
     fi
+    i=0
+    for devpath in ${device_paths_list[@]} ; do
+      if ! device_info=`echo "$output_sds" | grep -A 2 "$devpath"`  ; then
+        echo "ERROR: Not found expected $devpath device path on $sds_name."
+        echo "$output_sds"
+        return 1
+      else
```
Comment (Andrey-mp) [inline juju/fuel/functions]: use 'elif'

**Comment by Andrey-mp** [comment]
add 'g' option to sed's string.
sed 's/X/Y/g'

File: juju/fuel/functions (hunk 1 above)
Comment (tikitavi) [inline juju/fuel/functions]: could you explain?

Comment (Andrey-mp) [inline juju/fuel/functions]: you create array and iterate over this array.
in bash you can iterate just over values (string with spaces) -
for x in A B C D ; do

File: juju/fuel/functions [hunk 7]
```diff
@@ -213,54 +213,61 @@ function check_storage_pool {
...
-  fi
+  for sp in $sp_list ; do
+    query_cmd="scli --query_storage_pool --protection_domain_name $pd --storage_pool_name $sp"
+    output=`juju ssh $mch "$login_cmd && $query_cmd" 2>/dev/null`
+    if echo "$output" | grep "$key_to_check" | grep -q "$expected_value" ; then
+      echo "INFO: Success. Parameter $key_to_check in storage pool $sp is $expected_value."
+    else
+      echo "ERROR: Parameter $key_to_check in storage pool $sp is in wrong state."
+      echo "$output"
+      return 1
```
Comment (Andrey-mp) [inline juju/fuel/functions]: please check all pools and then return error

File: juju/fuel/functions [hunk 8]
```diff
@@ -213,54 +213,61 @@ function check_storage_pool {
...
   login_cmd="scli --login --username $username --password $password --approve_certificate"
-  query_cmd="scli --query_storage_pool --protection_domain_name $pd --storage_pool_name $sp"
-  output_storage_pool=`juju ssh $mch "$login_cmd && $query_cmd" 2>/dev/null | grep "Capacity alert thresholds"`
-  if echo "$output_storage_pool" | awk '{print$5}' | grep -q "$high_alert" ; then
-    if echo "$output_storage_pool" | awk '{print$7}' | grep -q "$critical_alert" ; then
-      echo "INFO: Success. Capacity alert thresholds: high is $high_alert, critical is $critical_alert"
+  for sp in $sp_list ; do
+    query_cmd="scli --query_storage_pool --protection_domain_name $pd --storage_pool_name $sp"
+    output_storage_pool=`juju ssh $mch "$login_cmd && $query_cmd" 2>/dev/null | grep "Capacity alert thresholds"`
+    if echo "$output_storage_pool" | awk '{print$5}' | grep -q "$high_alert" ; then
```
Comment (Andrey-mp) [inline juju/fuel/functions]: we talked about two separate blocks for checking these parameters

File: juju/fuel/functions [hunk 9]
```diff
@@ -213,54 +213,61 @@ function check_storage_pool {
...
+      if echo "$output_storage_pool" | awk '{print$7}' | grep -q "$critical_alert" ; then
+        echo "INFO: Success. Capacity alert thresholds: high is $high_alert, critical is $critical_alert"
+      else
+        echo "ERROR: Wrong capacity critical alert thresholds"
+        echo "$output_storage_pool"
+        return 1
```
Comment (Andrey-mp) [inline juju/fuel/functions]: the same about several pools - check all and then return error

File: juju/fuel/deploy-fuel.sh [hunk 10]
```diff
@@ -27,28 +26,32 @@ check_capacity_alerts 1 '80' '90'
...
 set_fuel_options password='Other_password'
 set_fuel_options zero-padding='true'
 set_fuel_options checksum-mode='true'
 set_fuel_options scanner-mode='true'
 set_fuel_options spare-policy='15'
 set_fuel_options capacity-high-alert-threshold='79'
 set_fuel_options capacity-critical-alert-threshold='89'
-set_fuel_options cached-storage-pools='sp'
-set_fuel_options rfcache-devices=$rfcache_path
+set_fuel_options cached-storage-pools='sp2'
```
Comment (Andrey-mp) [inline juju/fuel/deploy-fuel.sh]: if you set only one of two pools here then it should be checked

File: juju/fuel/functions [hunk 11]
```diff
@@ -332,12 +350,24 @@ function check_sp_name {
...
-      echo "INFO: Success. Storage pool name on $sds_name is $sp_name."
     fi
+    i=0
+    for devpath in $device_paths_list ; do
+      if ! device_info=`echo "$output_sds" | grep -A 2 "$devpath"`  ; then
+        echo "ERROR: Not found expected $devpath device path on $sds_name."
+        echo "$output_sds"
+        return 1
+      elif ! echo "$device_info" | grep -q ${sp_list[$i]} ; then
+        echo "ERROR: Wrong storage pool on $devpath on $sds_name."
```
Comment (alexey-mr) [inline juju/fuel/functions]: why dont add info about expected storage pool in error message.

File: juju/fuel/functions [hunk 12]
```diff
@@ -213,75 +213,135 @@ function check_storage_pool {
   key_to_check=$2
   expected_value=$3
 
-  sp=`get_config fuel-master storage-pools`
+  storage_pools=`get_config fuel-master storage-pools`
```
Comment (Andrey-mp) [inline juju/fuel/functions]: btw, good practice -
do not pass value that can be got from config

**Pull request merged**